src/proxy/outbound.rs

```diff
@@ -357,16 +434,69 @@ impl OutboundConnection {
             }
             debug!("built request as passthrough; no upstream found");
             return Ok(Request {
-                protocol: Protocol::TCP,
+                protocol: OutboundProtocol::TCP,
                 source: source_workload,
                 hbone_target_destination: None,
                 actual_destination_workload: None,
                 intended_destination_service: None,
                 actual_destination: target,
                 upstream_sans: vec![],
+                final_sans: vec![],
             });
         };
 
+        // Check whether we are using an E/W gateway and sending cross network traffic
+        if let Some(ew_gtw) = &us.workload.network_gateway {
+            if us.workload.network != source_workload.network {
```

Comment:
IIUC this means that if we find a workload in another network but it does NOT have a gateway, we will just send to the IP, which will not work (or worse, may go to some actual destination unrelated to the target). Do we have something protecting against this

Reply:
I can add it, but I don't think we ever did this check